api/grpc_server.go

```diff
@@ -102,6 +103,17 @@ func (s SpacemeshGrpcService) Broadcast(ctx context.Context, in *pb.BroadcastMes
 	return &pb.SimpleMessage{Value: "ok"}, nil
 }
 
+func (s SpacemeshGrpcService) BroadcastPoet(ctx context.Context, in *pb.BinaryMessage) (*pb.SimpleMessage, error) {
+	log.Info("GRPC Broadcast PoET msg")
+	err := s.Network.Broadcast(activation.PoetProofProtocol, in.Data)
+	if err != nil {
+		log.Warning("failed to broadcast PoET message")
```

Comment:
should be an error, not warning

Reply:
Yes, I agree, don't know why I made this a warning, but...

I was curious to see what could make this fail and discovered that there is no code-path that returns an error here. Don't get me wrong, errors could occur, but they are all swallowed for reasons beyond me. This should be considered tech-debt, imo (relevant to processing all gossip messages).

Reply:
specifically the issue is that this method doesn't return errors: https://github.com/spacemeshos/go-spacemesh/blob/develop/p2p/gossip/protocol.go#L196

cc: @y0sher